Cargo.toml

```diff
@@ -63,3 +63,9 @@ crate-type = ["cdylib", "rlib"]
 [profile.release]
 lto = true
 codegen-units = 1
+
+[patch.crates-io]
+datafusion = { git = "https://github.com/apache/datafusion.git", branch = "branch-50" }
+datafusion-substrait = { git = "https://github.com/apache/datafusion.git", branch = "branch-50" }
+datafusion-proto = { git = "https://github.com/apache/datafusion.git", branch = "branch-50" }
+datafusion-ffi = { git = "https://github.com/apache/datafusion.git", branch = "branch-50" }
```

Comment:
Thanks @timsaucer for the PR. `datafusion 50` has now been released, so I think this is now ready.